cinder/tests/unit/volume/drivers/lightos/test_lightos_storage.py

```diff
@@ -57,6 +57,54 @@ def factory(protocol, root_helper, driver=None,
         return None
 
 
+class DBMock(object):
+
+    def __init__(self):
+        self.data = {
+            "projects"
+        }
+
+    def get_or_create_project(self, project_name):
+        project = self.data["projects"].setdefault(project_name, {})
+        return project
+
+    def get_project(self, project_name, create_if_not_exist=True):
+        project = self.data["projects"].get(project_name, None)
+        return project if project else None
+
+    def delete_project(self, project_name):
+        project = self.get_project(project_name)
+        if not project:
+            return None
+        self.data["projects"].remove(project)
+
+    def create_volume(self, volume):
+        assert volume.project_name and volume.name, "must be provided"
+        project = self.get_or_create_project(volume.project_name)
+        volumes = project.setdefault("volumes", {})
```

Comment:
create volume in lightOS is not idempotent we will return error if already exist